### PR TITLE
Docs: Multihash

### PR DESCRIPTION
**Pull request by LiraLemur**
Signed-off-by: Sara <lira.lemur@gmail.com>

### Description of the Change

Some multihash docs

### Benefits

Some data on multihash in Iroha

### Possible Drawbacks 

Might not be enough but something to start with

### CODE REVIEW

File: docs/source/develop/keys.rst [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+Some accounts (if `quorum <../concepts_architecture/glossary.html#quorum>`_ is more than 1) might have more `Signatories <../concepts_architecture/glossary.html#signatory>`_ that sign transactions – and each Signatory has a pair of keys.
+Cryptographic algorithms use those keys – and in Iroha we provide you with a choice – which algorithms to use.
+
+Supported Crypto Algorithms
+===========================
+
+Natively, HL Iroha uses a custom algorithm – SHA-3 ed25519.
+These keys are supported by all versions of Iroha, including the old ones.
+But as we all know, we need more universal options as well – that is why Iroha has `HL Ursa integration <../integrations/index.html#hyperledger-ursa>`_ – it is a library with different crypto algorithms, that allows to work with Iroha using more mainstream keys.
+Ursa provides Iroha with support of standard SHA-2 ed25519 algorithm.
```
Comment (MBoldyrev) [inline docs/source/develop/keys.rst:15]:
```suggestion
Ursa provides Iroha with support of standard Ed25519 with SHA-2 algorithm.
```

File: docs/source/develop/keys.rst [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+
+To provide easy solution that would allow using different algorithms without "breaking" backward compatibility, we introduced **multihash** format in Iroha.
+You can learn more about multihash `here <https://github.com/multiformats/multihash>`_.
+
+Generally, to create keys, different from the native SHA-3 ed25519 keys, you will need to bring them to this format:
+
+.. code-block:: shell
+
+	<varint hash function code><varint digest size in bytes><hash function output>
```
Comment (MBoldyrev) [inline docs/source/develop/keys.rst:25]:
```suggestion
	<varint key type code><varint key size in bytes><actual key bytes>
```

I would also merge it with the example below

```
code    | length   | data 
--------+----------+-------------------- 
ed25519 | 32 bytes | ed25519 public key 
--------+----------+-------------------- 
ED      | 20       | 6264646435383430... 

```

File: docs/source/develop/keys.rst [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+
+Binary example from the multihash documentation (only 4 bytes for simplicity):
+
+.. code-block:: shell
+
+	fn code  dig size hash digest
+	-------- -------- ------------------------------------
+	00010001 00000100 101101100 11111000 01011100 10110101
+	sha1     4 bytes  4 byte sha1 digest
```
Comment (MBoldyrev) [inline docs/source/develop/keys.rst:38]: sha1 is irrelevant, pls use the data in the comment above

File: docs/source/develop/keys.rst [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+
+If Iroha receives a standard public key of 32 bytes, it will treat is as a native Iroha key.
+If it receives a multihash public key, it will treat it based on the table below.
```
Comment (MBoldyrev) [inline docs/source/develop/keys.rst:42]: I would add it to the table for illustrative clarity. like with empty first columns

File: docs/source/develop/keys.rst (hunk 3 above)
Comment (LiraLemur) [reply docs/source/develop/keys.rst:38]: got it

File: docs/source/develop/keys.rst (hunk 4 above)
Comment (LiraLemur) [reply docs/source/develop/keys.rst:32]: Kinda fixed?.. Could you please check the latest commit

File: docs/source/develop/keys.rst [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+But as we all know, we need more universal options as well – that is why Iroha has `HL Ursa integration <../integrations/index.html#hyperledger-ursa>`_ – it is a library with different crypto algorithms, that allows to work with Iroha using more mainstream keys.
+Ursa provides Iroha with support of standard Ed25519 with SHA-2 algorithm.
+
+Public Keys
+-----------
+
+To provide easy solution that would allow using different algorithms without "breaking" backward compatibility, we introduced **multihash** format for public keys in Iroha.
+You can learn more about multihash `here <https://github.com/multiformats/multihash>`_.
+
+Generally, to create keys, different from the native SHA-3 ed25519 keys, you will need to bring them to this format:
```
Comment (MBoldyrev) [inline docs/source/develop/keys.rst:23]:
```suggestion
Generally, to use keys, different from the native SHA-3 ed25519 keys, you will need to bring them to this format:
```

**Pull request merged**